python-threatexchange/threatexchange/ncmec/hash_api.py

```diff
@@ -0,0 +1,326 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Simple implementation for the NCMEC hash sharing XML API
+
+You can find the complete documentation at 
+https://report.cybertip.org/hashsharing/v2/documentation.pdf
+"""
+
+import xml.etree.ElementTree as ET
+from datetime import datetime
+from dataclasses import dataclass
+from enum import Enum, unique
+import logging
+import time
+import typing as t
+from io import BytesIO
+
+import requests
+from requests.packages.urllib3.util.retry import Retry
+
+# Maybe move to a common library someday
+from threatexchange.fb_threatexchange.api import TimeoutHTTPAdapter
+
+
+_DATE_FORMAT_STR = "%Y-%m-%dT%H:%M:%SZ"
+_DEFAULT_ELE = ET.Element("")
+
+T = t.TypeVar("T")
+
+
+def nullthrows(v: t.Optional[T]) -> T:
+    """Typing helper"""
+    assert v is not None
+    return v
+
+
+class _XMLWrapper:
+    """
+    Simpler wrapper around XML element for typing and null checking.
+
+    To use the raw methods, just use .element
+    """
+
+    def __init__(self, e: ET.Element) -> None:
+        self.element = e
+
+    def __getitem__(self, key) -> "_XMLWrapper":
+        """Asserts child with name exists"""
+        child = self.element.find(key)
```

Comment:
possible dumb question to help my understanding: why `find` not `findall`

Reply:
synced offline: should be the same because we expect well formed xml we can treat like json (warning if we are wrong this fails silently) so might be worth a todo/fix

Reply:
Converted to a 'splody version